plugins/destination/csv/client/write.go

```diff
@@ -0,0 +1,114 @@
+package client
+
+import (
+	"context"
+	"encoding/csv"
+	"os"
+	"path"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+func (c *Client) releaseTables(tables []string) {
+	for _, t := range tables {
+		if c.writers[t].count == 0 {
+			panic("unexpected count is 0")
+		}
+		c.writers[t].count--
+		if c.writers[t].count == 0 {
+			c.writers[t].writer.Flush()
+			fileName := c.writers[t].file.Name()
+			if err := c.writers[t].file.Close(); err != nil {
+				c.logger.Error().Err(err).Msgf("failed to close file %s", fileName)
+			}
+			delete(c.writers, t)
+		}
+	}
+}
+
+func (c *Client) startWrite(tables schema.Tables) error {
+	// we want to have the ability to rollback in case of an error so we dont have zombie files
+	var initializedTables []string
```

Comment:
why not keep `initializedTables` as a map inside `Client`? could be useful in `endWrite` too.

Reply:
this wont work as it can't be access concurrently and will introduce lock contention